src/openag_brain/software_modules/recipe_handler.py

```diff
@@ -105,6 +108,43 @@ def __iter__(self):
             yield (time.time(), variable, value)
         yield (time.time(), RECIPE_END.name, self.id)
 
+def hrs_to_seconds(hrs):
+    return hrs * (60 * 60)
+
+def days_to_seconds(days):
+    return hrs_to_seconds(days * 24)
+
+@interpret_recipe.register("phased")
+class PhasedRecipeInterpreter:
+    def __init__(self, recipe, start_time=None, timeout=1):
+        init_time = rospy.get_time()
+        start_time = start_time or init_time
+        if start_time > init_time:
+            raise ValueError("Recipes cannot be scheduled for the future")
+        self.start_time = start_time
+        self.stages = recipe["operations"]
+        self.id = recipe["_id"]
+        self.timeout = timeout
+
+    def __iter__(self):
+        yield (rospy.get_time(), RECIPE_START.name, self.id)
+        for stage in self.stages:
+            stage_duration = days_to_seconds(stage["days"])
+            end_of_stage = (rospy.get_time() - self.start_time) + stage_duration
```

Comment:
I'm probably just being pedantic here, but this computes the end time for the current stage relative to the current time instead of relative to when the previous stage was supposed to end. I believe the latter semantics would be more "correct". So basically I'm suggesting initializing `end_of_stage` to `self.start_time` outside of the loop and then adding `stage_duration` to it on every iteration. With the current implementation, stage start times are expected to drift forward from where you would naturally expect them to be by about `self.timeout/2` seconds per phase. This probably won't matter at all for most people tbh, but it's also a relatively easy change.